src/configs/passport.config.js

```diff
@@ -0,0 +1,90 @@
+const passport = require("passport");
+const LocalStrategy = require("passport-local").Strategy;
+const GoogleStrategy = require("passport-google-oidc");
+
+const userAccountService = require("../services/user/user.service");
+const passwordUtils = require("../utils/password.utils");
+const crypto = require("crypto");
+
+const envConfig = require("./env.config");
+
+//Local strategy to authenticate the user signing in
+passport.use(
+  new LocalStrategy(
+    {
+      usernameField: "email",
+      passwordField: "password",
+    },
+    async (email, password, done) => {
+      const user = await userAccountService.getUserAccount(email);
+      if (!user) {
+        return done(null, false, { message: "Invalid email" });
```

Comment:
`done` callback should throw a proper error if there's some error in validating the user-provided credentials. Under current implementations, if the user tries to sign in using some invalid user credentials, the server responds with the `401 Unauthorized` response rather than giving a proper error message.
Also, read about the [error-first callbacks](https://nodejs.org/en/learn/asynchronous-work/javascript-asynchronous-programming-and-callbacks#:~:text=the%20first%20parameter%20in%20any%20callback%20function%20is%20the%20error%20object%3A%20error%2Dfirst%20callbacks) to understand how errors are generally handled in the Node.js ecosystem.